components/google-cloud/storage/download/component.yaml

```diff
@@ -17,15 +18,30 @@ implementation:
 
             uri="$0"
             output_path="$1"
-
-            # Checking whether the URI points to a single blob, a directory or a URI pattern
-            # URI points to a blob when that URI does not end with slash and listing that URI only yields the same URI
-            if [[ "$uri" != */ ]] && (gsutil ls "$uri" | grep --fixed-strings --line-regexp "$uri"); then
+            default_data_path="$2"
+            
+            # Checking whether the URI exists
+            # TODO: Distinguish between NotFound error and other errors
+            if gsutil ls "$uri"; then
+                # Checking whether the URI points to a single blob, a directory or a URI pattern
```

Comment:
The order of this "if ... elif ..." implies:
1. if `GCS path` points somewhere nonexistent, it is okay as long as `Default data`, as a backup, exists.
2. Even if `Default data` is provided, it might not be used if `GCS path` is a valid. So the way to activate this `Default data` is to give `GCS path` some dummy value.

I'm trying to understand what would be a practical use-case example for this?

Reply:
>So the way to activate this Default data is to give GCS path some dummy value.

It's not intended to be activated on purpose. It's like a `catch` clause in `try/catch`.

>I'm trying to understand what would be a practical use-case example for this?

This feature is used in the [continue_training_from_prod](https://github.com/kubeflow/pipelines/blob/0795597562e076437a21745e524b5c960b1edb68/samples/core/continue_training_from_prod/continue_training_from_prod.py#L60) pipeline.

It tries to download the URI pointing to the production model that may not exist yet. In that case, the presence of `Default data` allows the component not to fail, but return something else as a fallback.